python/cugraph/cugraph/tests/centrality/test_betweenness_centrality_mg.py

```diff
@@ -33,158 +31,118 @@ def setup_function():
     gc.collect()
 
 
-IS_DIRECTED = [True, False]
+# =============================================================================
+# Parameters
+# =============================================================================
 
+DATASETS = [karate, dolphins]
+IS_DIRECTED = [True, False]
+IS_NORMALIZED = [True, False]
+ENDPOINTS = [True, False]
+SUBSET_SEEDS = [42, None]
+SUBSET_SIZES = [None, 15]
+VERTEX_LIST_TYPES = [list, cudf]
 
 # =============================================================================
-# Pytest fixtures
+# Helper functions
 # =============================================================================
 
-datasets = utils.DATASETS_UNDIRECTED
-
-fixture_params = gen_fixture_params_product(
-    (datasets, "graph_file"),
-    ([False, True], "normalized"),
-    ([False, True], "endpoints"),
-    ([42, None], "subset_seed"),
-    ([None, 15], "subset_size"),
-    (IS_DIRECTED, "directed"),
-    ([list, cudf], "vertex_list_type"),
-)
-
-
-@pytest.fixture(scope="module", params=fixture_params)
-def input_combo(request):
-    """
-    Simply return the current combination of params as a dictionary for use in
-    tests or other parameterized fixtures.
-    """
-    parameters = dict(
-        zip(
-            (
-                "graph_file",
-                "normalized",
-                "endpoints",
-                "subset_seed",
-                "subset_size",
-                "directed",
-                "vertex_list_type",
-            ),
-            request.param,
-        )
+
+def get_sg_graph(dataset, directed):
+    dataset.unload()
```

Comment:
I'm curious why we need to do this here and other places where we call it before getting a graph obj.

Reply:
These are needed because dataset objects persist their `self.edgelist` at the first time `get_edgelist` is called. Therefore, `unload` must be called if we're switching from SG usage to MG usage in order for the `dataset` object to store the correct type of edge list for its current use.

As a side-note, the weird bug I was seeing in CI was also related to the fact that certain `dataset` objects did not properly unload their internal edge-lists, so in subsequent tests the methods being called on `self.edgelist` were returning various errors.

Perhaps down the line, a more elegant solution can be implemented to remove the need for so many calls to `unload` within tests. Perhaps a flag (similar to how algorithms have `perform_expensive_check` where edgelists aren't persisted within the dataset object)

Reply:
Thanks, Ralph.  Good detective work tracking that down!